edge-hub/core/src/Microsoft.Azure.Devices.Routing.Core/checkpointers/Checkpointer.cs

```diff
@@ -35,6 +35,8 @@ public class Checkpointer : ICheckpointer
             this.closed = new AtomicBoolean(false);
             this.EndpointId = endpointId;
             this.Priority = priority;
+
+            Metrics.SetQueueLength((ulong)Math.Max(this.Offset, 0), endpointId, priority);
```

Comment:
This does not seem correct. The Queue length would be Queue.TailPointer - Checkpointer.Offset.

Reply:
Both the offset and proposed are set to the CheckpointData.Offset. This is also all there is on the checkpointData (Offset, LastFailedRevivalTime and UnhealthySince). Does this not map to what you would expect?

Reply:
This idea of using createAsync cannot work... It must be the case that if we have 10 messages all unprocessed... then the offset is 0. If this value was 10 then on propose we would get 11 - 10 for count of metric and in logs but that is not the case.

We also don't want to put in 'propose' call as that leads to same problem as we have now... let's discuss this a bit, as we need a new strategy. What might be the 'easiest' thing to do with very few changes... would to have checkpointer passed into MessageStore or have something exposed on Checkpointer to where we can take the count of messages from cleanup and 'adjust' the proposed and offset values. As the problem with current implementation is TTL where Checkpointer has no idea about messages that just disappear after 300 seconds (or whatever value). Still let's talk about this for bit and think this through :)